internal/gatewayapi/resource.go

```diff
@@ -20,6 +20,37 @@ import (
 
 type XdsIRMap map[string]*ir.Xds
 type InfraIRMap map[string]*ir.Infra
+type GatewayClassResources map[string]*Resources
```

Comment:
non blocking comment that can be tackled in a new PR - would a list `*Resources[]` be a better option here ?